intercept/src/intercept.h

```diff
@@ -2374,40 +2457,62 @@ inline bool CLIntercept::checkDumpImageEnqueueLimits(
             "Unmap", enqueueCounter, memobj, command_queue, NULL, 0, 0 );   \
     }
 
-#define DUMP_BUFFERS_BEFORE_ENQUEUE( kernel, command_queue )                \
+#define DUMP_BUFFERS_BEFORE_ENQUEUE( kernel, command_queue)                 \
     if( pIntercept->config().DumpBuffersBeforeEnqueue &&                    \
         pIntercept->checkDumpBufferEnqueueLimits( enqueueCounter ) &&       \
         pIntercept->dumpBufferForKernel( kernel ) )                         \
     {                                                                       \
         pIntercept->dumpBuffersForKernel(                                   \
-            "Pre", enqueueCounter, kernel, command_queue );                 \
-    }
-
-#define DUMP_BUFFERS_AFTER_ENQUEUE( kernel, command_queue )                 \
-    if( pIntercept->config().DumpBuffersAfterEnqueue &&                     \
-        pIntercept->checkDumpBufferEnqueueLimits( enqueueCounter ) &&       \
-        pIntercept->dumpBufferForKernel( kernel ) )                         \
-    {                                                                       \
-        pIntercept->dumpBuffersForKernel(                                   \
-            "Post", enqueueCounter, kernel, command_queue );                \
-    }
+            "Pre", enqueueCounter, kernel, command_queue, false, false );   \
+    }
+
+#define DUMP_BUFFERS_AFTER_ENQUEUE( kernel, command_queue )                     \
+    if(( pIntercept->config().DumpBuffersAfterEnqueue &&                        \
+        pIntercept->checkDumpBufferEnqueueLimits( enqueueCounter ) &&           \
+        pIntercept->dumpBufferForKernel( kernel )) ||                           \
+        (hasDumpedBufferByName && !hasDumpedValidationBufferByName ))           \
+    {                                                                           \
+        hasDumpedValidationBufferByName = true;                                 \
+        pIntercept->dumpBuffersForKernel(                                       \
+            "Post", enqueueCounter, kernel, command_queue, false,               \
+                pIntercept->config().DumpReplayKernelName != "" );              \
+    }                       
+
+#define DUMP_REPLAYABLE_KERNEL( kernel, command_queue, work_dim, gws_offset, gws, lws )                     \
+    if (enqueueCounter == static_cast<size_t>(pIntercept->config().DumpReplayKernelEnqueue ) ||             \
+        ( pIntercept->getShortKernelName(kernel) == pIntercept->config().DumpReplayKernelName &&            \
+        !hasDumpedBufferByName ))                                                                           \
```

Comment:
Couple of comments here:

1. It might be nice to combine these conditions into a "check" function like `checkDumpBufferEnqueueLimits` or `checkAubCaptureKernelSignature`.
2. If we do this I think we could keep `getShortKernelName` protected instead of making it public?
3. FWIW, the aubcapture check (for `AubCaptureKernelName`) is currently looking at the "long" kernel name and not the "short" kernel name.  It would be nice to be consistent.  I don't have a strong opinion whether this switches to the "long" kernel name or whether aubcapture switches to the "short" kernel name.

Reply:
Agreed on the first 2 parts, can also stay private instead.

No strong opinions either. I assume the short name is due to the Windows limitation on path lengths, so maybe the short kernel name is better?

Reply:
I am going to mark this one as resolved, as the first part is now implemented!

There seems to be more places where the long name is being used/checked (e.g. dumpBufferForKernel), with a comment asking if it should be the short name instead. Probably better for a separate PR!